tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -12,889 +13,945 @@
+        }
 
-            ParameterInfo[] onMethodBeginParameters = onMethodBeginMethodInfo.GetParameters();
-            if (onMethodBeginParameters.Length < argumentsTypes.Length)
-            {
-                ThrowHelper.ThrowArgumentException($"The method: {BeginMethodName} with {onMethodBeginParameters.Length} parameters in type: {integrationType.FullName} has less parameters than required.");
-            }
-            else if (onMethodBeginParameters.Length > argumentsTypes.Length + 1)
-            {
-                ThrowHelper.ThrowArgumentException($"The method: {BeginMethodName} with {onMethodBeginParameters.Length} parameters in type: {integrationType.FullName} has more parameters than required.");
-            }
-            else if (onMethodBeginParameters.Length != argumentsTypes.Length && onMethodBeginParameters[0].ParameterType != genericArgumentsTypes[0])
+        var onMethodBeginParameters = onMethodBeginMethodInfo.GetParameters();
+        if (onMethodBeginParameters.Length < argumentsTypes.Length)
+        {
+            ThrowHelper.ThrowArgumentException($"The method: {BeginMethodName} with {onMethodBeginParameters.Length} parameters in type: {integrationType.FullName} has less parameters than required.");
+        }
+        else if (onMethodBeginParameters.Length > argumentsTypes.Length + 1)
+        {
+            ThrowHelper.ThrowArgumentException($"The method: {BeginMethodName} with {onMethodBeginParameters.Length} parameters in type: {integrationType.FullName} has more parameters than required.");
+        }
+        else if (onMethodBeginParameters.Length != argumentsTypes.Length && onMethodBeginParameters[0].ParameterType != genericArgumentsTypes[0])
+        {
+            ThrowHelper.ThrowArgumentException($"The first generic argument for method: {BeginMethodName} in type: {integrationType.FullName} must be the same as the first parameter for the instance value.");
+        }
+
+        var callGenericTypes = new List<Type>();
+        var mustLoadInstance = onMethodBeginParameters.Length != argumentsTypes.Length;
+        var instanceGenericType = genericArgumentsTypes[0];
+        var instanceGenericConstraint = instanceGenericType.GetGenericParameterConstraints().FirstOrDefault();
+        Type? instanceProxyType = null;
+        if (instanceGenericConstraint != null)
+        {
+            var result = DuckType.GetOrCreateProxyType(instanceGenericConstraint, targetType);
+            instanceProxyType = result.ProxyType;
+            if (instanceProxyType is null)
             {
-                ThrowHelper.ThrowArgumentException($"The first generic argument for method: {BeginMethodName} in type: {integrationType.FullName} must be the same as the first parameter for the instance value.");
+                ThrowHelper.ThrowArgumentException($"The instance proxy type for method: {BeginMethodName} in type: {integrationType.FullName} is null.");
```

Comment:
When would/could we throw this? 🤔 This only happens if there's a duck typing failure right? So we should never actually make it into this block? 

Looking at `CreateTypeResult`, I _think_ it could only happen if `_proxyType` is `null` _and_ `_exceptionInfo` is `null`... but I can't work out if that ever happens...

I wonder if we can improve the annotations on `CreateTypeResult` somehow, as currently it's all quite ambiguous. Not for this PR, just thinking out loud

Reply:
True I think we can do that in other PR.